apps/platform/hooks/useTwoFactorModal.tsx

```diff
@@ -173,4 +173,30 @@ const TwoFactorModal = (props: Props) => {
   );
 };
 
-export default TwoFactorModal;
+export const useTwoFactorModal = () => {
+  const [openModal, setOpenModal] = useState(false);
+
+  const { twoFactor } = trpc.useContext();
+  const { data: session } = useSession();
+  const user = session?.user as UserType;
+
+  // Function to wrap mission-critical function calls with two-factor auth
+  async function withTwoFactorAuth(fn: Function) {
+    twoFactor.checkTwoFactorState
```

Comment:
You can instead do `trpc.twoFactor.checkTwoFactorState({  }, { enabled: condition }) ` and simply use the `async` and `await` instead of `then`

Reply:
Ok Chetan, thanks.